File: ClassicalAndLondonSchools/ClassicalAndLondonSchoolsTest/CustomerLondonStyleTests.cs
using ClassicalAndLondonSchools;
using Moq;

namespace ClassicalAndLondonSchoolsTest
{
    [TestClass]
    public class CustomerLondonStyleTests
    {
        private Customer _customer;

        [TestInitialize]
        public void Initialize()
        {
            _customer = new Customer();
        }

        [TestMethod]
        public void TestPurchaseShouldSucceedWhenInventoryIsEnough()
        {
            Mock<IStore> storeMock = new Mock<IStore>();
            storeMock.Setup(s => s.HasEnoughInventory(Products.Shampoo, 5)).Returns(true);
            bool isPurchased = _customer.Purchase(storeMock.Object, Products.Shampoo, 5);
            Assert.IsTrue(isPurchased);
            storeMock.Verify(s => s.HasEnoughInventory(Products.Shampoo, 5), Times.Once);
        }

        [TestMethod]
        public void TestPUrchaseShouldFailWhenInventoryIsNotEnough()
        {
            Mock<IStore> storeMock = new Mock<IStore>();
            storeMock.Setup(s => s.HasEnoughInventory(Products.Shampoo, 5)).Returns(false);
            bool isPurchased = _customer.Purchase(storeMock.Object, Products.Shampoo, 5);
            Assert.IsFalse(isPurchased);
            storeMock.Verify(s => s.RemoveInventory(Products.Shampoo, 5), Times.Never);
        }

        [TestCleanup]
        public void Cleanup()
        {
            _customer = null;
        }
    }
}

File: classical-and-london-schools-of-testings-java/src/tests/CustomerLondonStyleTests.java
package tests;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mockito;

import junit.framework.TestCase;
import store.Customer;
import store.Products;
import store.Store;

public class CustomerLondonStyleTests extends TestCase {
    private Customer customer;

    @Before
    public void setUp() {
        this.customer = new Customer();
    }

    @Test
    public void testPurchaseShouldSucceedWhenInventoryIsEnough() {
        Store store = Mockito.mock(Store.class);
        Mockito.when(store.hasEnoughInventory(Products.Shampoo, 5)).thenReturn(true);
        boolean isPurchased = this.customer.purchase(store, Products.Shampoo, 5);
        assertTrue(isPurchased);
        Mockito.verify(store, Mockito.times(1)).hasEnoughInventory(Products.Shampoo, 5);
    }

    @Test
    public void testPurchaseShouldFailWhenInventoryIsNotEnough() {
        Store store = Mockito.mock(Store.class);
        Mockito.when(store.hasEnoughInventory(Products.Shampoo, 5)).thenReturn(false);
        boolean isPurchased = this.customer.purchase(store, Products.Shampoo, 5);
        assertFalse(isPurchased);
        Mockito.verify(store, Mockito.times(0)).removeInventory(Products.Shampoo, 5);
    }

    @After
    public void tearDown() {
        this.customer = null;
    }
}
